File: Codeforces/377/A.cpp
#include <bits/stdc++.h>

using namespace std;


int total, visitados;
char mapa[505][505];
int n,m,k;


void bfs(pair<int,int> inicio){

    queue<pair<int,int>> fila;
    fila.push(inicio);
    int visitados = 0;
    int dx[] = {-1,1,0,0};
    int dy[] = {0,0,-1,1};
    visitados++;
    mapa[inicio.first][inicio.second] = '1';
    while(k < (total - visitados)){
        pair<int, int> atual = fila.front();
        fila.pop();
        int i = atual.first;
        int j = atual.second;

        for(int x=0; x<4; x++){
            if(k == total - visitados){
                return;
            }
            if(mapa[i+dx[x]][j+dy[x]] == '.'){
                mapa[i+dx[x]][j+dy[x]] = '1';
                fila.push({i+dx[x], j+dy[x]});
                visitados++;
            }
        }
    }
}



int main(){
    cin >> n >> m >> k;

    memset(mapa, 'x', sizeof(char) * 505 * 505 );

    char temp;

    pair<int,int> inicio;
    for(int i=1; i<=n; i++){
        for(int j=1; j<= m; j++){
            cin >> temp;
            mapa[i][j] = temp;
            if(temp == '.'){
                inicio = {i,j};
                total++;
            }
        }
    }


    bfs(inicio);

    for(int i=1; i<=n; i++){
        for(int j=1; j<= m; j++){
            if(mapa[i][j] == '.'){
                cout << 'X';
            } else if(mapa[i][j] == '1'){
                cout << '.';
            } else {
                cout << mapa[i][j];
            }

        }
        cout << endl;
    }


}

File: Codeforces/377/A.py
def bfs(i, j):
    global k
    visitado[i][j] = 1
    queue = [(i,j)]
    k-=1
    matriz[i][j] = '1'
    while queue:
        i,j = queue.pop(0)
            
        for x in [[0,1],[1,0],[0,-1],[-1,0]]:
            if 0 <= i+x[0] < n and 0 <= j+x[1] < m and matriz[i+x[0]][j+x[1]] == '.' and not visitado[i+x[0]][j+x[1]] and k:
                k -= 1
                visitado[i+x[0]][j+x[1]] = 1
                matriz[i+x[0]][j+x[1]] = '1'
                queue.append((i+x[0],j+x[1]))

n,m,need = [int(i) for i in input().split()]
visitado = [[0]*m for _ in range(n)]


matriz = []
inicio = (0,0)
c = 0
for i in range(n):
    matriz.append(list(input()))
    c += matriz[i].count('.')
k = c-need

for i in range(n):
    for j in range(m):
        if matriz[i][j] == '.':
            inicio = (i,j)
            break
    else:
        continue
    break

bfs(inicio[0], inicio[1])

for i in matriz:
    for j in i:
        if j == '1':
            print('.',end='')
        elif j == '.':
            print('X',end='')
        else:
            print(j, end='')
    print()

